client-src/elements/chromedash-typeahead.ts

```diff
@@ -130,6 +131,38 @@ export class ChromedashTypeahead extends LitElement {
     );
   }
 
+  containsOp(s: string | null): boolean {
+    if (s === null) {
+      return false;
+    }
+    const COMPARE_OPS = ['=', ':', '<', '>'];
+    return COMPARE_OPS.some(op => s.includes(op));
+  }
+
+  groupCandidates(candidates: Candidate[]): Candidate[] {
+    const groupsSeen = new Set();
+    const groupsSeenTwice = new Set();
+    for (const c of candidates) {
+      if (groupsSeen.has(c.group)) {
+        groupsSeenTwice.add(c.group);
+      } else {
+        groupsSeen.add(c.group);
+      }
+    }
```

Comment:
Optional nit, this could use a [destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment)
```suggestion
    for (const {group} of candidates) {
      if (groupsSeen.has(group)) {
        groupsSeenTwice.add(group);
      } else {
        groupsSeen.add(group);
      }
    }
```

Reply:
Tempting, but it feels too more natural to me to have `c` as the loop variable that holds each candidate.  I might do destructuring if the loop body lines were getting excessively complex or repetitive.